lib/phoenix/transports/long_poll_server.ex

```diff
@@ -112,6 +112,11 @@ defmodule Phoenix.Transports.LongPoll.Server do
     end
   end
 
+  def handle_info({:graceful_exit, channel_pid, %Phoenix.Socket.Message{} = msg}, state) do
+    new_state = delete(state, msg.topic, channel_pid)
```

Comment:
If we go that route then the EXIT will still propagate to the transport and potentially cause failures depending on how the EXIT is handled. For example, it would cause the long poll server to terminate today.